### ISSUE TITLE
Different styles and weights from same font family

### ISSUE THREAD

**Issue by johanbaaij** [mileszs/wicked_pdf#442]
I'm trying to use some variations of Roboto: Condensed reg, condensed light and condensed light italic to be exact.

I embeded the .ttf's in the stylesheet with the asset_data_base64 as mentiond in issue 250. When it's just one font everything works fine, but once I add another one it won't display multiple ones.

If I add a font from a different "family", for instance Open Sans, it works without a problem! This makes me think there must be a trick for getting this to work, I'm currently trying to rename the fonts to different families with FontForge but no luck so far.

Here's my relevant CSS;

```css
@font-face {
  font-family: "Roboto Condensed";
  font-style: normal;
  font-weight: 400;
  src:url("<%= asset_data_base64('robotocondensed-regular.ttf') %>");  
}

@font-face {
  font-family: "Roboto Condensed Light";
  font-style: normal;
  font-weight: 300;
  src:url("<%= asset_data_base64('robotocondensed-light.ttf') %>");  
}

@font-face {
  font-family: "Roboto Condensed Light";
  font-style: italic;
  font-weight: 300;
  src:url("<%= asset_data_base64('robotocondensed-lightitalic.ttf') %>");  
}

.front-primary {
  font-family: "Roboto Condensed";
}

.front-secondary {
  font-family: "Roboto Condensed Light";
}

.back-secondary {
  font-family: "Roboto Condensed Light";
  font-style: italic;
}
```

Am I missing something?

**Comment by unixmonkey** [comment]
This looks like it could help:

https://github.com/wkhtmltopdf/wkhtmltopdf/issues/2435

Try renaming your different `font-family` names to not share any matching words and/or be less than 8 chars.

Maybe RobotoC & RobotoCL?

I'm not sure if you'll need to actually rename the font itself, but maybe.

Let me know how it goes!

**Comment by johanbaaij** [comment]
Thanks, I did make some progress! I renamed all the fonts in FontForge and generated new .ttf's. Now the condensed regular and condensed light are working properly. The italic displays as the condensed light though, even though it displays correctly on it's own. I used random < 8 char names for the fonts and replaced every name field mentioning Roboto with the new name.

Quite strange, will try a few more things and let you know if I get it to work, thanks again

**Comment by unixmonkey** [comment]
Cool, thanks for the update!

**Comment by johanbaaij** [comment]
Wow, it seems to be working now. I changed the font style name for the italic one to Regular in FontForge then added them to my CSS like this;

```css
@font-face {
  font-family: "fontthree";
  font-style: normal;
  font-weight: 300;
  src:url("<%= asset_data_base64('fontthree.ttf') %>");  
}
.back-secondary {
  font-family: "fontthree";
  font-style: italic;
}
```

At first I thought it might have been a fake italic version of the condensed light but I can't tell the difference when I look at them next to each other.

**Comment by gwalshington** [comment]
This [StackOverflow](https://stackoverflow.com/questions/36427422/wkhtmltopdf-multiple-font-faces-styles/45524377#45524377) helped me find a solution. 

TLDR - you have to remove the 'Preferred Family' in the font files.

Hope it saves someone else a few hours...

**Comment by mapreal19** [comment]
@gwalshington it did thanks. That solution fixed the issue for me too
